workspace-manager-clienttests/src/main/java/scripts/testscripts/GetWorkspace.java

```diff
@@ -0,0 +1,90 @@
+package scripts.testscripts;
+
+import bio.terra.testrunner.runner.TestScript;
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.client.ApiClient;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.client.ApiException;
+import bio.terra.workspace.model.CreateWorkspaceRequestBody;
+import bio.terra.workspace.model.CreatedWorkspace;
+import bio.terra.workspace.model.WorkspaceDescription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scripts.utils.WorkspaceManagerServiceUtils;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class GetWorkspace extends TestScript {
+    private static final Logger logger = LoggerFactory.getLogger(GetWorkspace.class);
+    private UUID id;
+    private CreatedWorkspace workspace;
+
+    /** Public constructor so that this class can be instantiated via reflection. */
+    public GetWorkspace() {
+        super();
+    }
+
+    public void setup(List<TestUserSpecification> testUsers) throws Exception {
+        assertThat("There must be at least one test user in configs/testusers directory.", testUsers!=null && testUsers.size()>0);
+        id = UUID.randomUUID();
+        ApiClient apiClient = WorkspaceManagerServiceUtils.getClientForTestUser(testUsers.get(0), server);
+        WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
+        try {
+            CreateWorkspaceRequestBody requestBody = new CreateWorkspaceRequestBody();
+            requestBody.setId(id);
+            requestBody.setJobId("testrunner");
+            requestBody.setSpendProfile("testrunner");
+            workspace = workspaceApi.createWorkspace(requestBody);
+        } catch (ApiException apiEx) {
+            logger.debug("Caught exception creating workspace ", apiEx);
+        }
+
+        int httpCode = workspaceApi.getApiClient().getStatusCode();
+        logger.info("CREATE workspace HTTP code: {}", httpCode);
+        assertThat(httpCode, equalTo(200));
+    }
+
+    public void userJourney(TestUserSpecification testUser) throws Exception {
+        ApiClient apiClient = WorkspaceManagerServiceUtils.getClientForTestUser(testUser, server);
+        WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
+
+        /**
+         * This GetWorkspace test expects a valid workspace id
+         * created from the setup step.
+         *
+         * Throw workspace not found exception if anything goes wrong
+         * **/
+        try {
+            // String invalidWorkspaceId = "11111111-1111-1111-1111-111111111111";
+            WorkspaceDescription workspaceDescription = workspaceApi.getWorkspace(workspace.getId());
+            assertThat("GET workspace does not throw not found exception", true);
+        } catch (ApiException apiEx) {
```

Comment:
Also, is it possible that other exceptions (apart from "not found") can be thrown here, in which case the assert message below may be incorrect?
`assertThat("GET workspace throws not found exception", true);`

Reply:
It is up to team to catch application level error as they see fit. The current code is just an example on how to use test runner library.